Makefile

```diff
@@ -69,18 +78,38 @@ fix: ## Fix linting issues in the code
 	${DOCKER_COMPOSE} run --rm app npm run lint:fix
 
 test: export TARGET = dev
-test: ## Run the tests
+test: ## Run all the tests
+	$(MAKE) db
+	$(MAKE) wait-healthy
+	$(MAKE) initdb
 	${DOCKER_COMPOSE} run --rm app npm run test
 
+unit-test: export TARGET = dev
+unit-test: ## Run the unit tests
+	${DOCKER_COMPOSE} run --rm app npm run test:unit
+
+integration-test: export TARGET = dev
+integration-test: ## Run the integration tests
+	$(MAKE) db
+	$(MAKE) wait-healthy
+	$(MAKE) initdb
+	${DOCKER_COMPOSE} run --rm app npm run test:integration
+
 run:
 	${DOCKER_COMPOSE} up --abort-on-container-exit --exit-code-from app; ${DOCKER_COMPOSE} down
 
 dev: export TARGET = dev
 dev: ## Build and runs the container for development
 	$(MAKE) --jobs=4 install build stop
+	$(MAKE) db
+	$(MAKE) wait-healthy
+	$(MAKE) initdb
 	$(MAKE) run
 
 prod: export TARGET = prod
 prod: ## Builds and runs the container for production
 	$(MAKE) --jobs=2 build stop
+	$(MAKE) db
+	$(MAKE) wait-healthy
+	$(MAKE) initdb
 	$(MAKE) run
```

Comment:
Should be able to combine these into one (`$(MAKE) db wait-healthy initdb run`), as without `--jobs` they'll run in order.

Reply:
Though given the duplication, they should be part of `run` itself.

Reply:
And maybe combine `db`, `wait-healthy` and `initdb` into a single command.

Reply:
See no harm in pushing them in `run`.
Not sure if $(MAKE) propagates a `--jobs` down the recursive call?

Reply:
Yes, asking for parallelism on the command line propagates:
```
$ cat Makefile 
.PHONY: first second

first:
        sleep 1
        echo first

second:
        sleep 1
        echo second

parent:
        $(MAKE) first second
$ time make --jobs=2 parent
make first second
make[1]: Entering directory '/home/giorgio/code/make-experiments'
sleep 1
sleep 1
echo first
echo second
first
second
make[1]: Leaving directory '/home/giorgio/code/make-experiments'

real    0m1.014s
user    0m0.006s
sys     0m0.012s
```